src/main/domain/consent/LoadConsentService.js

```diff
@@ -74,34 +75,43 @@ export class LoadConsentService {
       return true
     }
 
+    const oldVendorList = await this._vendorListRepository.getVendorList({
+      version: new Version(consent.vendorListVersion)
+    })
+
     return (
       (await this._consentHaveConsentsAndLIForAllVendorsWithTheSameValue({
         consent,
         newVendorList,
-        value: true
+        value: true,
+        oldVendorList: oldVendorList.value.vendors
       })) ||
       this._consentHaveConsentsAndLIForAllVendorsWithTheSameValue({
         consent,
         newVendorList,
-        value: false
+        value: false,
+        oldVendorList: oldVendorList.value.vendors
       })
     )
   }
 
   async _consentHaveConsentsAndLIForAllVendorsWithTheSameValue({
     consent,
     newVendorList,
-    value
+    value,
+    oldVendorList
   }) {
     let isValid = false
     if (
       this._vendorListHelper.haveAllValuesTo({
         object: consent.vendor.consents,
-        valueToVerify: value
+        valueToVerify: value,
+        oldVendorList
       }) &&
       this._vendorListHelper.haveAllValuesTo({
         object: consent.vendor.legitimateInterests,
-        valueToVerify: value
+        valueToVerify: value,
```

Comment:
I think that it won't work with legitimateInterests beacuse IAB's encoder saves it to "false" no matter if we sent a "true" signal